### DISCUSSION TITLE
(PRO) Repeatable jQuery-selector error on some field-name's

### DISCUSSION THREAD

**Discussion by stefankuijt** [Laravel-Backpack/community-forum#713] [Q&A (Help)] [answered]
Hello!

I'm missing (fieldname) escaping in the repeatable-field javascript.

So i can't use the name component[test] as it will give an error because of the square bracket not being escaped.

I copied the file:
vendor/backpack/pro/resources/views/fields/repeatable.blade.php
To:
resources/views/vendor/backpack/crud/fields/repeatable.blade.php 

**And did the following changes.**

Replace:
`var field_name = element.attr('name');`
With:
`var field_name = element.attr('name');`
`var escaped_field_name = $.escapeSelector(field_name);`

Replace:
`$('input[type=hidden][name='+field_name+']').first().on('CrudField:disable', function() {`
With:
`$('input[type=hidden][name='+escaped_field_name+']').first().on('CrudField:disable', function() {`

Replace:
`$('input[type=hidden][name='+field_name+']').first().on('CrudField:enable', function() {`
With:
`$('input[type=hidden][name='+escaped_field_name+']').first().on('CrudField:enable', function() {`

Replace:
`$('input[type=hidden][name='+$(repeatable).attr('data-repeatable-holder')+']').first().trigger('change', [values]);`
With:
`var escaped_field_name = $.escapeSelector($(repeatable).attr('data-repeatable-holder'))`
`$('input[type=hidden][name='+ escaped_field_name +']').first().trigger('change', [values]);`

**Would love to see them added in the repository so i can remove my copy of the file and use the default one**

**Comment by pxpm** [comment, marked as answer]
Hey @stefankuijt thanks for the report. 

We were already aware of the [issue here](https://github.com/Laravel-Backpack/community-forum/issues/238) and we postponed it almost one year ago and never got back to it. 😞 

The reason we didn't prioritized is because we can't guarantee that all kind of selectors will work, even if escaped, so you will also probably need to overwrite some backpack save/update process if deviating from the "standard process".

We'd agree to escape the selectors back then, so I've done it already in PRO and tagged a new version 2.0.17. 

It should be available later today with a `composer update backpack/pro`

Closing this, let me know if you still experiencing issues after the update. 

Cheers

**Reply by stefankuijt** [reply]
I will check later, thanks for the fast reaction/fix!
